---
title: "Supplementary material of New insights of the RNa-seq data of The Cancer Genome Atlas for Papillary Thyroid Carcinoma"
output:
  html_document:
    theme: united
    toc: yes
    toc_depth: 4
    toc_float:
      collapsed: false
      smooth_scroll: false
  pdf_document:
    toc: yes
---
```{r setup, cache = TRUE, echo = FALSE, results = 'asis'}
library("knitr")
# dumpcssfile <- function(fname) {
#   paste(c('<style type = "text/css">', readLines(fname), '</style>\n'),
#         collapse = "\n")
# }
# 
# cat(dumpcssfile(file.path("css", "ieo.css")))

knitr::opts_chunk$set(autodep = TRUE,
               fig.align = "center",
               comment = "",
               tidy = TRUE,
               message = FALSE,
               source = function(x, options) {
    paste("\\begin{lstlisting}[numbers=left, firstnumber=last]\n", x, 
        "\\end{lstlisting}\n", sep = "")
}) 

# Modifications based on https://rpubs.com/ajlyons/autonumfigs
# A function for generating captions and cross-references
fig <- local({
    i <- 0
    list(
        ref=function(refName, link=FALSE, checkRef=TRUE) {
            
            ## This function puts in a cross reference to a caption. You refer to the
            ## caption with the refName that was passed to fig$cap() (not the code chunk name).
            ## The cross reference can be hyperlinked.
            
            if (checkRef && !refName %in% names(ref)) stop(paste0("fig$ref() error: ", refName, " not found"))
            if (link) {
                paste0("<A HREF=\"#", refName, "\">Figure ", ref[[refName]], "</A>")
            } else {
                paste0("Figure ", ref[[refName]])
            }
        },
        cap=function(refName, text, center=FALSE, col="black", inline=FALSE) {
            i <<- i + 1
            ref[[refName]] <<- i
            css_ctr <- ""
            if (center) css_ctr <- "text-align:center; display:inline-block; width:100%;"
            cap_txt <- paste0("<span style=\"color:", col, "; ", css_ctr, "\">Figure ", i, ": ", text , "</span>")
            anchor <- paste0("<a name=\"", refName, "\"></a>")
            if (inline) {
                paste0(anchor, cap_txt)    
            } else {
                list(anchor=anchor, cap_txt=cap_txt)
            }
        },
        ref_all=function(){
            ## For debugging
            ref
        })
})

## This chunk replaces the default hook for processing plots. It achieves the purposes,
## of laying out auto-numbered captions, but other functionality may be gone.

knit_hooks$set(plot = function(x, options) {
    sty <- ""
    if (options$fig.align == 'default') {
        sty <- ""
    } else {
        sty <- paste0(" style=\"text-align:", options$fig.align, ";\"")
    }
    
    if (is.list(options$fig.cap)) {
        ## options$fig.cap is a list returned by the function fig$cap()
        str_caption <- options$fig.cap$cap_txt
        str_anchr <- options$fig.cap$anchor
    } else {
        ## options$fig.cap is a character object (hard coded, no anchor)
        str_caption <- options$fig.cap
        str_anchr <- ""
    }
    
    paste('<figure', sty, '>', str_anchr, '<img src="',
        opts_knit$get('base.url'), paste(x, collapse = '.'),
        '"><figcaption>', str_caption, '</figcaption></figure>',
        sep = '')
    
})

## This chucnk will read through *this* Rmd file, and attempt to extract all of the 
## labels (not caption text) used for Figure captions. These labels are used
## as anchors, so scanning through the document now will allow us to create cross references
## before the caption actually appears. 

## Get the name of this Rmd file
rmdFn <- knitr::current_input()  # filename of input document

## Read lines and close connection
rmdCon <- file(rmdFn, open = "r")
rmdLines <- readLines(rmdCon)
close(rmdCon)

## Pull out all occurences of at least one back tick, followed 
## by any number of characters, followed by fig$cap (all on one line)
figscap_idx <- grep("`+(.*)fig\\$cap", rmdLines)
rmdLines <- rmdLines[figscap_idx]

## Get rid of everything up until the start of the caption label
## This presumes the caption label is the first argument of fig$cap()
## E.g., fig.cap = "\\label{my_label", ...)
rmdLinesSansPre <- sub("(.*)fig\\$cap(.*?)[\"']", "", rmdLines)

## Identify everything up until the first quote
match_data <- regexpr("(.*?)[\"']", rmdLinesSansPre)

## Reduce the length by one, because we're not interested in the final quote
attr(match_data, "match.length") <- attr(match_data, "match.length") - 1

## Extract
fig_labels <- regmatches(rmdLinesSansPre, match_data, invert=FALSE)

if (length(fig_labels) > 0) {

    ## Test for duplicates
    if (anyDuplicated(fig_labels) > 0) stop("Duplicate caption labels detected")
    
    ## Create a named list of Figure numbers
    ref <- as.list(1:length(fig_labels))
    names(ref) <- fig_labels
}    
```

#### Mauro Álvarez (mauro.alvarez01@estudiant.upf.edu)
#### Lluís Revilla (lluis.revilla01@estudiant.upf.edu)
#### Marina Reixachs (marina.reixachs01@estudiant.upf.edu)
#### Inés Sentís (ines.sentis01@estudiant.upf.edu) 

## Introduction

Thyroid cancer can develop from the different cells that form the follicles of the thyroid. This gland located at the base of the throat secretes hormones such as T3 and T4 that have their metabolic functionalities such as control of heart rate, blood pressure, body temperature and weight. [See it on TCGA](http://cancergenome.nih.gov/cancersselected/thyroid). 

From the four types of thyroid cancer (papillary, follicular, medullary, and anaplastic thyroid cancer) Papillary Thyroid Carcinoma (PTC) is the most common type of thyroid cancer (Agrawal et al., 2014). It is more prevalent in women than men and its common diagnosis occurs around 49 years old.

Driver onco-mutations for this type of cancer appear as alterations of the MAPK signaling pathway and the PI3K-AKT pathway (Agrawal et al., 2014). Both are implied in cell proliferation and survival and in human tumorigenesis. The overactivation of the MAPK pathway because of mutations such as the BRAFV600E mutation, leads to the development of PTC from follicular thyroid cells (Xing, 2013). On the other hand, mutations that activate the PI3K-AKT pathway, such as mutations in RAS, PTEN and PIK3CA, mostly leads to development of Follicular Thyroid Adenoma (FTA) and Follicular Thyroid Cancer (FTC) also in follicular thyroid cells (Xing, 2013).

Research from The Cancer Genome Atlas [(TCGA)](http://cancergenome.nih.gov/cancersselected/thyroid) focused on PTC and they found that a part from alterations in BRAF (specifically BRAFV600E) and RAS, there are other driver mutations such the ones in EIF1AX PPM1D, and CHEK2 genes that are main alterations for that type of thyroid cancer(Agrawal et al., 2014).

## Data import

We start importing the raw table of counts.

```{r start, message=FALSE, autodep=TRUE}
library(SummarizedExperiment)

thca <- readRDS("seTHCA.rds") 
thca
```

### Data exploration

Explore the column (phenotypic) data, which in this case corresponds to clinical
variables, and their corresponding metadata.

```{r explore1, autodep=TRUE}
dim(colData(thca))
col.data <- colData(thca)
col.data[1:5, 1:5]

col.data.meta <- mcols(colData(thca), use.names=TRUE)
col.data.meta
```

Now, explore the row (feature) data.

```{r explore2, autodep=TRUE}
rowRanges(thca)
```

Looking deeper into col.data with:

```{r explore.col.data, autodep=TRUE}
col.data[1:5, 30:35]
```

We observed that there are different ways to indicate that the data is not available (NA, [Not Available]...). Consequently, we decide to study which variables have more information. To do so, we create two functions, one (`na.replace`) to change those values into the standarized `NA` nomenclature, and `filter.info` to quantify how many of each variable is not `NA`.

```{r functions, autodep=TRUE}

na.replace <- function(x){
    # Remove the unwanted factors
    lev <- levels(x)
    lev <- lev[!(lev %in% "[Not Available]")]
    lev <- lev[!(lev %in% "[Unknown]")]
    lev <- lev[!(lev %in% "[Not Applicable]")]
    lev <- lev[!(lev %in% "[Not Available]|[Not Available]|[Not Available]")]
    lev <- lev[!(lev %in% "[Not Evaluated]")]
    lev <- lev[!(lev %in% "None")]

    x <- factor(x, levels=lev)
}

filter.info <- function(x){
    # Function to filter the NA, [Not Available], [Not Applicable] and [Unknown]
    # Return the proportion of information on x
    nas <- sum(is.na(x))
    return(1 - (nas/length(x)))
}
```

We apply these functions to the `col.data` and we visualize it:

```{r data.analysis, fig.cap="\\label{fig:col.data.analysis}Figure 1. Information in each column", autodep=TRUE}
meta.info <- as.data.frame(lapply(col.data, na.replace))
variable.info <- sapply(meta.info, filter.info)

# Representing the information by column vs total information
relative.info <- variable.info[order(variable.info)]/sum(variable.info)
plot(relative.info, main="Information brought by column")
```

```{r col.data.analysis2, fig.cap="\\label{col.data.analysis2}Figure 2.A Histogram of information of additional information per sample.", autodep=TRUE}
# Plotting the histogram of the information
hist(variable.info, 
     main = "Variables completness", xlab = "% of not NA of each variable", ylab = "Counts")
```

```{r col.data.analysis3, fig.cap="\\label{\\label{col.data.analysis3}Figure 2.B: A zoom on the most informative variables.", autodep=TRUE}
# Seeing the most informative to set a threshold,
# frequency are the number of columns with such % of information
hist(variable.info, xlim = c(0.3, 1), ylim = c(0, 35), 
     main = "Variables completness", xlab = "% of not NA of each variable", ylab = "Counts")
```

As we can see, in many variables the main value is just `NA` or other derivatives meaning "No information available for this variable". Therefore, in order to keep the most informative variables, we apply a threshold of 60% (of the values containing informative data) to select the relevant ones:

```{r select.filtered, autodep=TRUE}
filtered <- meta.info[, variable.info  > 0.6]

dim(filtered)
meta.info.summary2 <- col.data.meta[variable.info  > 0.6,]
```

Thus we ended with `r ncol(filtered)` informatives variables.

### Associated variables exploration

To explore the variables associated with the samples we first create an object with `ggplot`:

```{r ggplot.config, autodep=TRUE}
library("ggplot2")
fplot <- ggplot(as.data.frame(filtered))

## ggplot options for axes and background color
# Options of the labels
l <- theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust = 1))
# Removing the background
b <-  theme(axis.line = element_line(colour = "black"),
            panel.grid.major.x = element_blank(),
            panel.grid.major.y = element_line(colour = "grey", 
                                              size = 0.5, linetype = 2),
            panel.grid.minor.y = element_line(colour = "grey", 
                                              size = 0.25, linetype = 3),
            panel.border = element_blank(),
            panel.background = element_blank())
y.axis <- ylab("number of cases")

```

And we plot some relationships for some significant variables:  
In the "normal" type samples most of the information is not available. This could be due to the fact that all "normal" samples were obtained from patients with tumors. So in order to have the maximum information of each sample, we infer "normal" samples information from its paired sample data (mainly gender and ethnicity).

```{r plots3A, fig.cap="\\label{gender}Figure 3.A: The gender distribution: threre are more females than males.", autodep=TRUE}
fplot + geom_bar(aes(type, fill = gender)) + b + y.axis + 
    ggtitle("Gender distribution across normal and tumour samples")
```

As expected, there are more female samples than male samples. As we mentioned before, thyroid cancer is more prevalent in women than men and we can see that not only for tumor samples but for normals samples too. In addition, there are more tumor cases than normal cases.

```{r plots3B,fig.cap="\\label{status}Figure 3.B: Some tumoral samples are not clearly assigned a tumor status.", autodep=TRUE}
fplot + geom_bar(aes(type, fill = tumor_status)) + b + y.axis + 
    ggtitle("Tumor status across normal and tumour samples")
```

We also have examined the tumor status of the tumor samples. There are more cancer patients without tumor ("tumor free") than with the tumor ("with tumor") after the collection of the data. We assume that most of the patients that participate in the study have recovered after it. 

```{r plots3C, fig.cap="\\label{ethnicity}Figure 3.C: The predominant ethnicity are not hispanic or latino", autodep=TRUE}
fplot + facet_wrap(~ gender) + geom_bar(aes(type, fill = ethnicity)) + b + 
    y.axis + ggtitle("Ethnicity distribution across normal and tumour samples")
```

In this plot we can see clearly the lack of data for some variables regarding normal samples. In other words, there is no record for normal samples neither females nor males about the hispanic origin of the individuals.

```{r plots3D, fig.cap="\\label{ethnicity2}Figure 3.D: Most samples are from white ancestry.", autodep=TRUE}
labels <- c("AM. INDIAN OR ALASKA NAT.", "ASIAN", "BLACK OR AFRICAN AM.", 
            "WHITE", "NA")
fplot + facet_wrap(~ ethnicity) + geom_bar(aes(race, fill = gender)) + b +
    l + y.axis + scale_x_discrete(breaks=waiver(), labels = labels) +
    ggtitle("Gender distribution across different ethnicity and race.")
```

If we have a look at the ethnicities, gender and hispanic origin, the data shows more cases for "white" individuals "not hispanic or latino" than "white" individuals "hispanic or latino".

```{r plots3E, fig.cap="\\label{race}Figure 3.E: Both in males and females the most abundant race is white.", autodep=TRUE}
fplot + geom_bar(aes(x=gender, fill = race)) + b + y.axis + 
    ggtitle("Ethnicity distribution in male and female")
```

If we ignore the hispanic origin of the individuals and just focus on the main ethnicities of the study we can observe that for both female and male samples there is  a majority of "white" individuals. However, the fact that "white" subjects are hispanic or not can provide some variability so that is something we will take into account later. 

```{r plots3F, fig.cap="\\label{focality}Figure 3.F: There are some tumors which we don't know the focality.", autodep=TRUE}
fplot + facet_wrap(~ type ) + geom_bar(aes(tumor_focality, fill = gender)) + b +
    l + y.axis + xlab("tumor focality") + 
    ggtitle("Gender distribution in different tumour focality")
```

There is no data about tumor focality in the normal samples, but this could be due to the fact that are samples obtained probably during the tumor removal or for a biopsy, from a healty part of the tyroid gland. For this reason there is no reason for the normal samples to have this feature.

```{r plots3G, fig.cap="\\label{diagnosis}Figure 3.G The most common diagnosticated tumor is Papillary thyroid Carcinoma.", autodep=TRUE}
labels <- c("Other specify", "Papillary - Classical", "Papillary - Follicular", 
            "Papillary - Tall Cell", "NA")
hd.xaxis <-  scale_x_discrete(name = "histologic diagnosis", breaks=waiver(), 
                labels=labels)
fplot + geom_bar(aes(histologic_diagnosis, fill = gender)) + b + l + hd.xaxis + 
    y.axis + 
    ggtitle("Gender distribution across different types of hystologic diagnosis")
```

Here, we can see that the tumor samples came mainly from classical Papillary Thyroid Carcinoma. We expect differential expression between the different thyroid cancer phenotypes so this might be a source of variability.

```{r plots3H, fig.cap="\\label{diagnosis2}Figure 3.H: There is a representative sample of diagnosis over the age of patients.", fig.height = 7, fig.width = 14, dpi = 100, echo = TRUE, autodep=TRUE}
labels <- c("Other specify", "Papillary - Classical", "Papillary - Follicular", 
            "Papillary - Tall Cell", "NA")
ggplot(filtered[!is.na(filtered$age_at_diagnosis), ]) + 
    geom_bar(aes(age_at_diagnosis, na.rm = TRUE, fill = histologic_diagnosis)) +
    b + l + labs(x="age at diagnosis") + y.axis + 
    scale_fill_discrete(name = "histologic diagnosis", breaks = waiver(), 
    labels = labels) + 
    ggtitle("Histologic diagnosis distribution across different ages")
```

Even thought, thyroid cancer is commontly diagnosed at mature ages (~50) this plot shows that the classical papillary thyroid cancer has similar representation in all ages for this dataset.

As a summary from the previous plots, the most abundant population in the samples is the "white" and "not hispanic or latino", so we decide to perform the analysis on this group of individuals, whose tumoral sample correspond to the most common cancer in thyroids: Papillary Thyroid Carcinoma(PTC). 

### Subseting

With this approach we want to avoid to include in the analysis, samples that introduce variability (mainly differential expression due to enviromental or epigenetic factors) as much as possible. To have more insight on the disease we want to have just paired samples, that is, samples from the same participant. So we filter the data accordingly:

```{r subset, autodep=TRUE}
# Paired data
paired <- intersect(thca[, thca$type == 'tumor']$bcr_patient_barcode, 
                    thca[, thca$type == 'normal']$bcr_patient_barcode)
paired_mask <- thca$bcr_patient_barcode %in% paired

filtered_filt <- filtered[paired_mask, ]

# Imposing the selected conditions
tumor_names <- row.names(subset(filtered_filt, 
    type == "tumor" &
    race == "WHITE" &
    histologic_diagnosis == "Thyroid Papillary Carcinoma - Classical/usual" &
    ethnicity == "NOT HISPANIC OR LATINO" ))

# Extracting the participants identifiers
participants <- unlist(lapply(tumor_names, substr, 9, 12))
participants <- participants[participants != ""]

check <- function(x, checklist){
    # Returning name of the sample just if the name is the same as in checklist
    a <- substr(x, 9, 12)
    if(a %in% checklist){
        return(x)
    }
}

# Extracting sample names of tumor and normal data.
sample_names <- unlist(lapply(row.names(filtered_filt), check, 
                              checklist = participants))
thca.filtered <- thca[, sample_names]
```

We are left with `r ncol(thca.filtered)` samples. We explore again the variables for this subset. In this case, we show just some of the relationship plots we have seen previously:

```{r subset.explore1, fig.cap="\\label{pairedg}Figure 4.A: The samples are consistent with the paired data.", autodep=TRUE}
filtered_filt <- filtered_filt[row.names(filtered_filt) %in% sample_names, ]
mplot <- ggplot(filtered_filt) + y.axis
mplot + geom_bar(aes(type, fill = gender)) + b +
  ggtitle("Ethnicity and type of sample after filtering")
```

This plot just checks whether we have the same number of cases in normal sample and tumor sample so that we have done correctly the filtering.

```{r subset.explore2, fig.cap="\\label{pairede}Figure 4.B: The ethnicity is also consistent with the filter applied", autodep=TRUE}
mplot + facet_wrap(~ gender) + geom_bar(aes(type, fill = ethnicity)) + b +
  ggtitle("Hispanic origin, gender and type of sample after filtering")
```

Hispanic origin data is only available for tumor samples but as we have mentioned before we can extrapolate ethinicity and origin to the normal samples because they are paired (see below). 

```{r subset.explore3, fig.cap="\\label{pairedge}Figure 4.C: The normal samples are perfectly correlated with the number of tumor samples, despite normal samples have 'NA' as ethnicity", autodep=TRUE}
mplot + geom_bar(aes(x=gender, fill = ethnicity)) + b + l +
  ggtitle("Gender samples and hispanic origin after filtering")
```

Same proportion of being or not being "hispanic or latino" in females and males. Subset filtering has not modified the fact that we have more female samples than males. 

```{r subset.explore4, fig.cap="\\label{paireda}Figure 4.D: This graph shows that the paired data filtering is consistent", autodep=TRUE}
mplot + geom_bar(aes(histologic_diagnosis, fill = gender)) + b + l + 
    scale_x_discrete(name = "histologic diagnosis", breaks=waiver(), 
    labels = c("Papillary - Classical", "NA")) + 
    ggtitle("Gender distribution across different types of hystologic diagnosis
            after filtering")
```

Papillary Classical Thyroid cancer in the same proportion as "NA" because we have the same number of normal cases and classical thyroid tumor cases. 

```{r subset.explore5, fig.cap="\\label{fig:subset.explore5}Figure 4.E: We have a epresentative sample of all ages.", autodep=TRUE}
ggplot(filtered_filt[!is.na(filtered_filt$age_at_diagnosis), ]) +
    geom_bar(aes(age_at_diagnosis, fill = histologic_diagnosis)) + b + l + y.axis +
    xlab("age at diagnosis") + scale_fill_discrete(name = "histologic diagnosis",
    breaks=waiver(), labels = c("Papillary - Classical", "NA")) + 
    ggtitle("Histologic diagnosis distribution across different ages after filtering")
```

Number of cases across ages is more or less similar after the filtering exept for some ages that have more cases around 30, 34, 42 and 47. 

### Transforming counts to rpkm

To perform quality assessment and normalization we need first to load the
[edgeR](http://bioconductor.org/packages/edgeR) R/Bioconductor package and
create a `DGEList' object.

```{r edgeR, message = FALSE, autodep=TRUE}
library(edgeR)
# Normalization just with the selected samples
dge.subset <- DGEList(counts = assays(thca.filtered)$counts, 
                      genes = as.data.frame(mcols(thca.filtered)))
dge <- DGEList(counts = assays(thca)$counts, genes = as.data.frame(mcols(thca)))
```

Now calculate $\log_2$ CPM values of expression and put them as an additional
assay element to ease their manipulation.

```{r logCPM, autodep=TRUE}
assays(thca.filtered)$logCPM <- cpm(dge.subset, log=TRUE, prior.count=3)
assays(thca.filtered)$logCPM[1:5, 1:5]
```

## Quality assessment and normalization

### Library sizes

Let's examine the library sizes in terms of total number of sequence read counts
per sample. Figure 5 below shows library sizes per sample in increasing order.

```{r libsizes1, echo=TRUE, out.width="600px", fig.cap="\\label{libsize1}Figure 5.A: Library sizes in increasing order of the whole dataset", autodep=TRUE}
ord <- order(dge$sample$lib.size)
barplot(dge$sample$lib.size[ord]/1e6, las=1, ylab="Millions of reads",
        xlab="Samples", col=c("blue", "red")[(thca$type[ord] == "tumor") + 1],
        border = NA)
legend("topleft", c("tumor", "normal"), fill=c("red", "blue"), inset=0.01)
```

Plotting for all samples makes it impossible to see accurately the blue and red colors 
corresponding to tumor and normal respectively.

```{r libsizes2, echo=TRUE, out.width="600px", fig.cap="\\label{libsize2}Figure 5.B: Library sizes in increasing order of the paired data.", autodep=TRUE}
ord.subset <- order(dge.subset$sample$lib.size)
barplot(dge.subset$sample$lib.size[ord.subset]/1e6, las=1, ylab="Millions of reads",
        xlab="Samples", col=c("blue", "red")[(thca.filtered$type[ord.subset] == "tumor") + 1],
        border = NA)
legend("topleft", c("tumor", "normal"), fill=c("red", "blue"), inset=0.01)
```

After the filtering the subset plot allows one to see that there is similar heterogeneity in tumor and normal library sizes. In addition, this figure reveals substantial differences in sequencing depth between samples and we may consider discarding those samples whose depth is substantially lower than the rest. To identify which are these samples we may simply look at the actual numbers including portion of the sample identifier that distinguishes them.

```{r libsize.set, fig.cap="\\label{libsizen}Figure 6: Distribution of library size compared with the normal distribution.", autodep=TRUE}
sampledepth <- round(dge.subset$sample$lib.size / 1e6, digits = 1)
names(sampledepth) <- substr(sample_names, 6, 12)

qqnorm(sampledepth)
qqline(sampledepth)
```

Looking into the qqplot we decide to filter out those below 40 but keeping only those that are paired:

```{r refiltering.names, autodep=TRUE}
sample_names <- sample_names[sampledepth > 40]

keep.paired <- function(pattern, x){
    # Return the names if it is paired
    pos <- grep(pattern, x, fixed = TRUE)
    if (length(pos) > 1){
        return(x[pos])
    }
}

sample_names <- unique(unlist(lapply(substr(sample_names, 9, 12), 
                                     keep.paired, sample_names)))

thca.filtered <- thca.filtered[, colnames(thca.filtered) %in% sample_names]
dge.subset <- DGEList(counts = assays(thca.filtered)$counts, 
                      genes = mcols(thca.filtered))
```

Thus we have `r sum(sampledepth > 40)` samples which are above the threshold and `r ncol(thca.filtered)` of them are paired, which are the ones which will be used from here on.

### Distribution of expression levels among samples

Let's look at the distribution of expression values per sample in terms of
logarithmic CPM units. Due to the large number of samples, we display tumor
and normal samples separately, and are shown in Figure 7.

```{r distRawExp, fig.height = 4, fig.width = 10, out.width = "800px", fig.cap="\\label{distRawExp}Figure 7: Non-parametric density distribution of expression profiles per sample.", message=FALSE, autodep=TRUE}
library(geneplotter)
par(mfrow = c(1, 2))
multidensity(as.list(as.data.frame(assays(thca.filtered[, 
            thca.filtered$type ==  "tumor"])$logCPM)),
            xlab = "log 2 CPM", legend = NULL, main = "Tumor samples", 
            las = 1)
multidensity(as.list(as.data.frame(assays(thca.filtered[,
            thca.filtered$type ==  "normal"])$logCPM)),
            xlab = "log 2 CPM", legend = NULL, main = "Normal samples",
            las = 1)
```

These two spikes are due to the lowly expressed genes and the highly expressed genes, but in the lowly expressed genes the effect of GC content and the read length may have higher impact.

### GC and length effect

To see the impact of GC and length on the number of read we plot them:

```{r gc_length, fig.cap="\\label{gc_length}Figure 8: Relationship between GC content and length with the mean expression of genes.", autodep=TRUE}
gc_len <- rowRanges(thca.filtered)
cor(rowMeans(assays(thca.filtered)$counts), gc_len$txgc)
cor(rowMeans(assays(thca.filtered)$counts), gc_len$txlen)

par(mfrow=c(1,2))
plot(log10(rowMeans(assays(thca.filtered)$counts)), gc_len$txgc, 
     main = "GC content over expression", xlab = "Expression (log10)", 
     ylab = "GC")
plot(log10(rowMeans(assays(thca.filtered)$counts)), log10(gc_len$txlen), 
     main = "Length over the expression", xlab = "Expression (log10)", 
     ylab = "Length (log10)")
```

There is a low correlation on the counts and the GC content or the length but as there are clearly two groups based on the length, so we will normalize also using the length with the package cqn:

```{r cqn, message=FALSE, autodep=TRUE, fig.cap="\\label{cqn}Figure 9: The effect of GC and length on each sample."}
library("cqn")
cqn.subset <- cqn(assays(thca.filtered)$counts, x = gc_len$txgc, lengths = gc_len$txlen)
par(mfrow = c(1, 2))
boxplot(cqn.subset$func1, names = FALSE, xlab = "Samples", 
        main = "Estimated effect of GC")
boxplot(cqn.subset$func2, names = FALSE, xlab = "Samples", 
        main = "Estimated effect of length")
```

As we thought the effect of the length is higher than the effect of GC content

We can see if the distribution of expression values per sample in terms of logarithmic CPM units. Due to the large number of samples, we display tumor and normal samples separately:

```{r distExp, fig.height = 4, fig.width = 10, out.width = "800px", fig.cap="\\label{distExp}Figure 10: Non-parametric density distribution of expression profiles per sample.", autodep=TRUE}
cqn.logCPM <- cqn.subset$y + cqn.subset$offset
assays(thca.filtered)$cqn.logCPM <- cqn.logCPM
par(mfrow = c(1, 2))
multidensity(as.list(as.data.frame(
    cqn.logCPM[,thca.filtered$type ==  "tumor"])),
     xlab = "log 2 CPM", legend = NULL, main = "Tumor samples", las = 1)
multidensity(as.list(as.data.frame(
    cqn.logCPM[, thca.filtered$type ==  "normal"])),
    xlab = "log 2 CPM", legend = NULL, main = "Normal samples", las = 1)
```

These "after-normalization" plots are quite better than the previous ones used without GC and length normalization on the second spike. With this normalization we do not appreciate substantial differences between the samples in the distribution of expression values. However, below 0 there is a lot of variability.

### Distribution of expression levels among genes

Let's calculate now the average expression per gene through all the samples. Figure 9
shows the distribution of those values across genes.

```{r exprdist, out.width = "400px", fig.cap = "\\label{exprdist}Figure 11: Distribution of average expression level per gene using DGE normalization and cqn normalization", autodep=TRUE}
par(mfrow = c(1, 2))
avgexp <- rowMeans(assays(thca.filtered)$logCPM)
hist(avgexp, xlab = "log2 CPM", main = "Using DGEList normalization", las = 1)
abline(v = 1, col = "red", lwd = 2)

# Using the corrected logCPM
avgexp.cqn <- rowMeans(assays(thca.filtered)$cqn.logCPM)
hist(avgexp.cqn, xlab = "log2 CPM", main = "Using cqn normalization", las = 1)
abline(v = 1, col = "red", lwd = 2)
```

### Filtering of lowly-expressed genes

In the light of previous plot, we may consider a cutoff of 1 log CPM unit as minimum value
of expression to select genes being expressed across samples. Using this cutoff we proceed
to filter out lowly-expressed genes.

```{r filter.low.expr, autodep=TRUE}
mask <- avgexp.cqn > 1
thca.filtered <- thca.filtered[mask, ]
dge.subset <- dge.subset[mask, ]
```

We have `nrow(thca.filtered)` genes remaining on our dataset from `r ncol(thca.filtered)` samples, half of them are healthy and the other half are tumoral.

### Normalization

Since we will use the offsets computed by `cqn`, there is no need to normalize using the normalization tools from `edgeR`, such as `calcNormFactors` according to cqn vignette.

```{r norm, fig.height = 16, fig.cap="\\label{norm}Figure 12: Tumor and Normal samples after filtering, without normalization, with calcNormFactors normalization and with cqn normalization", autodep=TRUE}
dge.subset$offset <- cqn.subset$glm.offset

dge.norm <- calcNormFactors(dge.subset)
assays(thca.filtered)$n.logCPM <- cpm(dge.norm, log = TRUE, prior.count = 3)

par(mfrow = c(3, 2))

# Not normalized but filtered
multidensity(as.list(as.data.frame(
    assays(thca.filtered[,thca.filtered$type ==  "tumor"])$logCPM)),
     xlab = "log 2 CPM", legend = NULL, main = "Tumor samples", las = 1)
multidensity(as.list(as.data.frame(
    assays(thca.filtered[,thca.filtered$type ==  "tumor"])$logCPM)),
    xlab = "log 2 CPM", legend = NULL, main = "Normal samples", las = 1)

# Normalize with calcNormFactors after filtering out those
multidensity(as.list(as.data.frame(
    assays(thca.filtered[,thca.filtered$type ==  "tumor"])$n.logCPM)),
     xlab = "log 2 CPM", legend = NULL, main = "Tumor samples", las = 1)
multidensity(as.list(as.data.frame(
    assays(thca.filtered[,thca.filtered$type ==  "tumor"])$n.logCPM)),
    xlab = "log 2 CPM", legend = NULL, main = "Normal samples", las = 1)

# Normalized with cqn and filtered the low values:
multidensity(as.list(as.data.frame(
    assays(thca.filtered[,thca.filtered$type ==  "tumor"])$cqn.logCPM)),
     xlab = "log 2 CPM", legend = NULL, main = "Tumor samples", las = 1)
multidensity(as.list(as.data.frame(
    assays(thca.filtered[,thca.filtered$type ==  "tumor"])$cqn.logCPM)),
    xlab = "log 2 CPM", legend = NULL, main = "Normal samples", las = 1)
```

The normalization after filtering lowly-expressed genes with 'cqn' package clusters samples into one thin gaussian curve whereas normalization with calcNormFactors makes it thiker. 

### MA-plots

#### Tumoral samples

We examine now the MA-plots of the normalized expression profiles. We look first to
the tumor samples.

<!---
Here we make a MA-plot for each sample. The options 'fig.height' and 'fig.width'
control the relative image size in *inches*. The final image size results from
'height'x'dpi' and 'width'x'dpi', where 'dpi' is the image resolution in
"dots per inch" (by default dpi = 72). To scale the image to a desired size use
'out.width' and 'out.height'. More information at http://yihui.name/knitr/options
--->

```{r maPlotsTumor, fig.height = 16, fig.cap = "\\label{maPlotsTumor}Figure 13.A: MA-plots of the tumor samples.", autodep=TRUE}
par(mfrow = c(8, 3), mar = c(4, 3, 4, 1))
setmp <- thca.filtered[, thca.filtered$type ==  "tumor"]
dgetmp <- dge.subset[, thca.filtered$type ==  "tumor"]
for (i in 1:ncol(setmp)) {
  A <- rowMeans(assays(setmp)$cqn.logCPM)
  M <- assays(setmp)$cqn.logCPM[, i] - A
  samplename <- substr(as.character(setmp$bcr_patient_barcode[i]), 1, 12)
  smoothScatter(A, M, main = samplename, las = 1)
  abline(h = 0, col = "blue", lwd = 2)
  lo <- lowess(M ~ A)
  lines(lo$x, lo$y, col = "red", lwd = 2)
}
```

We do not observe samples with major expression-level dependent biases. 

#### Healthy samples

Let's look now to the normal samples:

```{r maPlotsNormal, fig.height = 10, fig.cap = "\\label{maPlotsNormal}Figure 13.B: MA-plots of the normal samples.", autodep=TRUE}
par(mfrow = c(8, 3), mar = c(4, 3, 4, 1))
setmp <- thca.filtered[, thca.filtered$type ==   "normal"]
dgetmp <- dge.subset[, thca.filtered$type ==  "normal"]
for (i in 1:ncol(setmp)) {
  A <- rowMeans(assays(setmp)$cqn.logCPM)
  M <- assays(setmp)$cqn.logCPM[, i] - A
  samplename <- substr(as.character(setmp$bcr_patient_barcode[i]), 1, 12)
  smoothScatter(A, M, main = samplename, las = 1)
  abline(h = 0, col = "blue", lwd = 2)
  lo <- lowess(M ~ A)
  lines(lo$x, lo$y, col = "red", lwd = 2)
}
```

We do not observe either important expression-level dependent biases among the normal samples except on the sample TCGA-BJ-A3PR. Which we proceed to remove it from the cqn.logCP:

```{r remove_sample, autodep=TRUE}
mask <- -grep("TCGA.BJ.A3PR", colnames(thca.filtered))
thca.filtered <- thca.filtered[,mask]
sample_names <- sample_names[mask]
dge.subset <- dge.subset[,mask]
```

Thus the remaining  samples are `r ncol(thca.filtered)` from `r ncol(thca.filtered)/2` patients. Recall that we still have `r nrow(thca.filtered)` genes from each sample.

### Batch identification

We will search now for potential surrogate of batch effect indicators. Given that each sample
names corresponds to a TCGA barcode (see the [wiki](https://wiki.nci.nih.gov/display/TCGA/TCGA+barcode)), following the strategy described [here](http://bioinformatics.mdanderson.org/main/TCGABatchEffects:Overview) we are going to derive different elements of the TCGA barcode and examine their distribution
across samples.

```{r barcodes, autodep=TRUE}
tss <- substr(sample_names, 6, 7)
table(tss)
samplevial <- substr(sample_names, 14, 16)
table(samplevial)
portion <- substr(sample_names, 18, 19)
table(portion)
table(substr(sample_names, 20, 20))
plate <- substr(sample_names, 22, 25)
table(plate)
center <- substr(sample_names, 27, 28)
table(center)
```

We can observe that the samples where collected at 4 tissues source sites but the majority of them at University of Pittsburg (BJ) and MD Anderson(EL). However,the manager center of the data is the same (University of North Carolina) and the same metabolite (RNA). In addition, they come from 7 different plates more or less distributed. However, most samples came from 2 out of 5 different portion    and 2 out of 3 vials.

We are going to check if the previous variables of the barcode are surrogate of batch effect indicator. Considering our outcome
of interest as molecular changes between sample types, tumor vs. normal, we will examine now
the cross-classification of this outcome with the variables:

```{r table.barcodes, autodep=TRUE}
table(data.frame(TYPE = thca.filtered$type, TSS = tss))
# We will check for the other variables
table(data.frame(TYPE = thca.filtered$type, PLATE = plate))
table(data.frame(TYPE = thca.filtered$type, Portion = portion))
table(data.frame(TYPE = thca.filtered$type, Vial = samplevial))
```
From the tables we can already see that all of them can be candidates as indicators for batch effect as there is no homogeneity of the number of tumor and normal samples across any of the variables. 

To double check if they have some effect on the batch we make the hierarchical clustering of them. First, we set some functions to plot the hierarhcical clustering:

```{r clustering.func, autodep=TRUE}
help.dendogram <- function(x, batch, labels) {
    # Helper function to plot dendograms
    if (is.leaf(x)) {
        ## color by batch
        attr(x, "nodePar") <- list(lab.col = as.vector(batch[attr(x, "label")])) 
        ## label by outcome
        attr(x, "label") <- as.vector(labels[attr(x, "label")])
    }
    x
}

plot.batch <- function(dendrogram, batching, se, out){
    # Function to see the batch
    # Given a sampleDendogram use batching to colour the leafs by sample.names of se
    sample.names = colnames(se)
    batch <- as.integer(factor(batching))
    names(batch) <- sample.names
    i.dendrogram <- dendrapply(dendrogram, help.dendogram, batch, out)
    plot(i.dendrogram, main = "Hierarchical clustering")
    legend("topright", paste("Batch", levels(factor(batching))), fill = sort(unique(batch)))
}
```

With this functions to plot the clustering with different labelings is easier:

```{r clustering.n1, fig.height = 7, fig.width = 14, dpi = 100, echo = TRUE, fig.cap="\\label{clustering.n1}Figure 14.A: Hierarchical clustering of the samples by portion", autodep=TRUE}
d <- as.dist(1-cor(assays(thca.filtered)$cqn.logCPM, method = "spearman"))
sampleClustering <- hclust(d)
sampleDendrogram <- as.dendrogram(sampleClustering, hang = 0.1)
outcome <- paste(substr(colnames(thca.filtered), 9, 12),
                        as.character(thca.filtered$type), sep = "-")
names(outcome) <- colnames(thca.filtered)
par(mfrow = c(1,1))
plot.batch(sampleDendrogram, portion, thca.filtered, outcome)
```

```{r clustering.n2, fig.height = 7, fig.width = 14, dpi = 100, echo = TRUE, fig.cap="\\label{clustering.n2}Figure 14.B: Hierarchical clustering of the samples by sample vial", autodep=TRUE}
plot.batch(sampleDendrogram, samplevial, thca.filtered, outcome)
```

```{r clustering.n3, fig.height = 7, fig.width = 14, dpi = 100, echo = TRUE, fig.cap="\\label{clusering.n3}Figure 14.C: Hierarchical clustering of the samples by TSS", autodep=TRUE}
plot.batch(sampleDendrogram, tss, thca.filtered, outcome)
```

```{r clustering.n4, fig.height = 7, fig.width = 14, dpi = 100, echo = TRUE, fig.cap="\\label{clustering.n5}Figure 14.D: Hierarchical clustering of the samples by type", autodep=TRUE}
plot.batch(sampleDendrogram, as.character(thca.filtered$type), thca.filtered, outcome)
```

```{r clustering.n5, fig.height = 7, fig.width = 14, dpi = 100, echo = TRUE, fig.cap="\\label{clustering.n5}Figure 14.E: Hierarchical clustering of the samples by prospective collection", autodep=TRUE}
plot.batch(sampleDendrogram, plate , thca.filtered, outcome)
```

We can see that the sample from A3PR-normal clusters with the tumoral ones, and the A3H2 tumoral clusters with the healthy patients (where?). 
There might be a batch effect on the samplevial, that is the order of portion in a sequence of 100 - 120 mg sample portions (not clear). We can also observe that there are two groups of tumoral samples, and one of them clusters with one sample from a healthy person (indicate graph). We can see with a PCA that the tumoral and normal cells don't create two clearly separated groups:

```{r mds, fig.height = 7, fig.width = 14, dpi = 100, echo = TRUE, fig.cap = "\\label{mds}Figure 15: Multidimensional scaling plot of the samples.", autodep=TRUE}
batch_effects <- list(portion, tss, as.character(thca.filtered$type),
                      as.character(thca.filtered$gender))
for (batching in batch_effects){
    batch <- as.integer(factor(batching))
    names(batch) <- sample_names
    plotMDS(dge.subset, labels = outcome, col = batch)
    legend("bottomleft", paste("Batch", levels(factor(batching))),
           fill = sort(unique(batch)), inset = 0.05, cex=0.7)
}
```

We can confirm that there isn't a clear group of normal and tumors and some tumoral samples do mix with the normal ones. And it seems that the females have more variance between healthy and tumoral.

## Design of the effects on expression

For now on we need to select our design and null hypothesis. We define them as:

```{r design, autodep=TRUE, include=FALSE}
# Create the design matrix involving tumor and normal samples and males and females
treat <- factor(paste(thca.filtered$type, thca.filtered$gender, sep = "."))
paired <- as.numeric(as.factor(thca.filtered$bcr_patient_barcode))
design <- model.matrix(~treat + paired, data=colData(thca.filtered))
colnames(design) <- c(levels(treat), "paired")
# design <- cbind("(Intercept)" = 1, design)
head(design)

# And the null model
mod0 <- model.matrix(~1 + gender + type + paired, data = colData(thca.filtered))
head(mod0)
```


## Removing batch effect

### ComBat

After looking for batch effect we couldn't find a clear cause of the batch effect. We suspect that sample vial has something to do. However, we try to see if we can improve the clustering by using method ComBat for sample vial. 

```{r clustering.combat, message=FALSE, fig.height = 7, fig.width = 14, dpi = 100, echo = TRUE, fig.cap = "\\label{clustering.combat}Figure 16.A: Hierarchical clustering of the samples with ComBat correction.", autodep=TRUE}
library("sva")

combatexp <- ComBat(assays(thca.filtered)$cqn.logCPM, tss, mod = design)

d <- as.dist(1 - cor(combatexp, method = "spearman"))
cluster.combat <- hclust(d)
cluster.combat <- as.dendrogram(cluster.combat, hang = 0.1)
names(batch) <- colnames(thca.filtered)
outcome <- paste(substr(colnames(thca.filtered), 9, 12),
                            as.character(thca.filtered$type), sep = "-")
names(outcome) <- colnames(thca.filtered)

plot.batch(cluster.combat, samplevial, thca.filtered, outcome)
```

```{r clustering.combat2, message=FALSE, fig.height = 7, fig.width = 14, dpi = 100, echo = TRUE, fig.cap = "\\label{clustering.combat2}Figure 16.B: Hierarchical clustering of the samples with ComBat correction coloured by gender.", autodep=TRUE}
plot.batch(cluster.combat, thca.filtered$gender, thca.filtered, outcome)
```

```{r clustering.combat3, message=FALSE, fig.height = 7, fig.width = 14, dpi = 100, echo = TRUE, fig.cap ="\\label{clustering.combat3}Figure 16.C: Hierarchical clustering of the samples with ComBat coloured by type.", autodep=TRUE}
plot.batch(cluster.combat, thca.filtered$type, thca.filtered, outcome)
```

We can observe a slightly improvement, but still some tumoral samples cluster together with the normal ones. 

```{r mds2, fig.height = 7, fig.width = 14, dpi = 100, echo = TRUE, fig.cap = "\\label{mds}Figure 16: Multidimensional scaling plot of the samples.", autodep=TRUE}
batch_effects <- list(portion, tss, as.character(thca.filtered$type),
                      as.character(thca.filtered$gender))
for (batching in batch_effects){
    batch <- as.integer(factor(batching))
    names(batch) <- sample_names
    plotMDS(combatexp, labels = outcome, col = batch)
    legend("bottomleft", paste("Batch", levels(factor(batching))),
           fill = sort(unique(batch)), inset = 0.05, cex=0.7)
}
```

### QR Decomposition

To improve the batch effect removal we look if QR decomposition yields better results. In this case, we look for sample vial and TSS batch effects removal. To do so, we have to built the full model which would include gender and type as fixed effects. 

```{r clustering.QRdecomposition.1, fig.height = 7, fig.width = 14, dpi = 100, echo = TRUE, fig.cap="\\label{clustering.QRdecomposition.1}Figure 17.A: Hierarchical clustering of the samples with QR descompostion for sample vial", autodep=TRUE}
library("limma")

qrexp <- removeBatchEffect(assays(thca.filtered)$cqn.logCPM, tss, design = design)
d <- as.dist(1 - cor(qrexp, method = "spearman"))
cluster.qr <- hclust(d)
cluster.qr <- as.dendrogram(cluster.qr, hang = 0.1)

plot.batch(cluster.qr, samplevial, thca.filtered, outcome)
```

```{r clustering.QRdecomposition.2, fig.height = 7, fig.width = 14, dpi = 100, echo = TRUE, fig.cap="\\label{clustering.QRdecomposition.2}Figure 17.B: Hierarchical clustering of the samples with QR descompostion for TSS", autodep=TRUE}
plot.batch(cluster.qr, tss, thca.filtered, outcome)
```

```{r clustering.QRdecomposition.3, fig.height = 7, fig.width = 14, dpi = 100, echo = TRUE, fig.cap="\\label{clustering.QRdecomposition.3}Figure 17.C: Hierarchical clustering of the samples with QR descompostion for gender", autodep=TRUE}
plot.batch(cluster.qr, thca.filtered$gender, thca.filtered, outcome)
```

```{r clustering.QRdecomposition.4, fig.height = 7, fig.width = 14, dpi = 100, echo = TRUE, fig.cap="\\label{clustering.QRdecomposition.4}Figure 17.D: Hierarchical clustering of the samples with QR descompostion for type", autodep=TRUE}
plot.batch(cluster.qr, thca.filtered$type, thca.filtered, outcome)
plot.batch(cluster.qr, tss, thca.filtered, outcome)
```

With QR decomposition we can observe that some tumoral samples cluster with other normal samples on a lower branch.

```{r mds3, fig.height = 7, fig.width = 14, dpi = 100, echo = TRUE, fig.cap = "\\label{mds}Figure 16: Multidimensional scaling plot of the samples.", autodep=TRUE}
batch_effects <- list(portion, tss, as.character(thca.filtered$type),
                      as.character(thca.filtered$gender))
for (batching in batch_effects){
    batch <- as.integer(factor(batching))
    names(batch) <- sample_names
    plotMDS(qrexp, labels = outcome, col = batch)
    legend("bottomleft", paste("Batch", levels(factor(batching))),
           fill = sort(unique(batch)), inset = 0.05, cex=0.7)
}
```

### Removing batch effect with SVD

We further try if SVD yelds a better result:

```{r clustring.svd1, fig.height = 7, fig.width = 14, dpi = 100, echo = TRUE, fig.cap="\\label{clustering.svd1}Figure 18.A: Hierarchical clustering of the samples with SVD for sample vial", autodep=TRUE}
library("corpcor")
s <- fast.svd(t(scale(t(assays(thca.filtered)$cqn.logCPM),
                      center = TRUE, scale = TRUE)))
pcSds <- s$d
pcSds[1] <- 0
svdexp <- s$u %*% diag(pcSds) %*% t(s$v)
colnames(svdexp) <- colnames(thca.filtered)
class(svdexp)
dim(svdexp)

d <- as.dist(1 - cor(svdexp, method = "spearman"))
cluster.svd <- hclust(d)
cluster.svd <- as.dendrogram(cluster.svd, hang = 0.1)
names(batch) <- colnames(thca.filtered)

plot.batch(cluster.svd, samplevial, thca.filtered, outcome)
```

```{r clustring.svd2, fig.height = 7, fig.width = 14, dpi = 100, echo = TRUE, fig.cap="\\label{clustering.svd2}Figure 18.B: Hierarchical clustering of the samples with SVD for TSS", autodep=TRUE}
plot.batch(cluster.svd, tss, thca.filtered, outcome)
```

```{r clustring.svd3, fig.height = 7, fig.width = 14, dpi = 100, echo = TRUE, fig.cap="\\label{clustering.svd3}Figure 18.C: Hierarchical clustering of the samples with SVD for type", autodep=TRUE}
plot.batch(cluster.svd, thca.filtered$type, thca.filtered, outcome)
```

SVD clusters together normal and tumoral samples, maybe because some of them are quite similar according to the PCA previously performed.
After all the process the best one, that separates better tumor and normal samples is the ComBat function, which will be used for following-up analysis.

```{r qrexpr, autodep=TRUE}
assays(thca.filtered)$qrexp <- qrexp
assays(thca.filtered)$combatexp <- combatexp
```

## Initial assesment on differential expression

We perform a simple examination of expression changes and their associated p-values
using the R/Bioconductor package [sva](http://bioconductor.org/packages/sva).
We build the null model with just the intercept (=1) in order to compare it with the previous full model (including gender and type).

```{r sva.model, autodep=TRUE}
pv <- f.pvalue(assays(thca.filtered)$combatexp, design, mod0)
sum(p.adjust(pv, method = "fdr") < 0.01)
```

There are `r sum(p.adjust(pv, method = "fdr") < 0.01)` genes changing significantly
their expression at FDR < 1%. In Figure 18 below we show the distribution of the
resulting p-values.

```{r pdist, out.width = "500px", fig.cap ="\\label{pdist}Figure 19.A: Distribution of raw p-values for an F-test on every gene between tumor and normal samples.", autodep=TRUE}
hist(pv, main = "Histogram of p-values", las = 1)
```

```{r pdist2, out.width = "500px", fig.cap = "\\label{pdist2}Figure 18.B: Distribution of raw p-values for an F-test on every gene between tumor and normal samples.", autodep=TRUE}
hist(pv, main = "Histogram of p-values", xlim = c(0, 0.1), breaks = 80)
```

### Surrogate variables with sva

Now, let's estimate surrogate variables using the `sva()` function.

```{r sva.variables, autodep=TRUE}
sv <- sva(assays(thca.filtered)$combatexp, design, mod0)
```

The SVA algorithm has found `r sv$n` surrogate variables. Let's use them to
assess against the extent of differential expression this time adjusting for these
surrogate variables.

```{r svn.ftest, autodep=TRUE}
modsv <- cbind(design, sv$sv)
mod0sv <- cbind(mod0, sv$sv)
pvsv <- f.pvalue(assays(thca.filtered)$combatexp, modsv, mod0sv)
sum(p.adjust(pvsv, method = "fdr") < 0.01)
```

We have increased the number of changing genes to `r sum(p.adjust(pvsv, method = "fdr") < 0.01)`. So removing or taking into account the surrogate variables lead to more differentially expressed genes.  
Figure 20 shows the resulting distribution of p-values.

```{r psvdist, out.width = "500px", fig.cap = "\\label{psvdist}Figure 20.A: Distribution of raw p-values for an F-test on every gene between tumor and normal samples, adjusting for surrogate variables estimated with SVA.", autodep=TRUE}
hist(pvsv, main = "Histogram of p-values adjusting surrogate variables", 
     las = 1)
```

```{r psvdist2, out.width = "500px", fig.cap ="\\label{psvdist2}Figure 20.B: Distribution of raw p-values for an F-test on every gene between tumor and normal samples, adjusting for surrogate variables estimated with SVA.", autodep=TRUE}
hist(pvsv, main = "Histogram of p-values adjusting surrogate variables", 
     las = 1, xlim = c(0, 0.1), breaks = 80)
```

After this preliminary analysis, we can conclude that aproximatelly 4300 genes have different expression levels among tumor and normal samples. 

### Removing surrogate variables effect

Instead of working with the surrogate variables in the model, we remove them for commodity when making the different contrast we are interested in.

```{r remove_surrogates, autodep=TRUE}
# Code from https://www.biostars.org/p/121489/#121500

cleanY = function(y, mod, svs) {
    X = cbind(mod, svs)
    Hat = solve(t(X) %*% X) %*% t(X)
    beta = (Hat %*% t(y))
    rm(Hat)
    gc()
    P = ncol(mod)
    return(y - t(as.matrix(X[,-c(1:P)]) %*% beta[-c(1:P),]))
}

e.no_surrogates <- cleanY(assays(thca.filtered)$combatexp, design, sv$sv)
```

The object e.no_surrogates contains the expression after removing the surrogates variables.

## Functional Annotation

We want ensembl annotation, and gene symbol of our probes, and we want also the gene ontologies from all the human genes (will be later used for functional enrichment):

```{r annot, autodep=TRUE}
library("org.Hs.eg.db")
annot <- select(org.Hs.eg.db, 
                columns = c("SYMBOL", "ENSEMBL"),
                keys = rownames(thca.filtered),
                keytype = "ENTREZID")
all.Human.GO <- select(org.Hs.eg.db,
                       columns = c("GO", "SYMBOL"), 
                       key = keys(org.Hs.eg.db, keytype = "ENTREZID"), 
                       keytype = "ENTREZID")
annot.Entrez.GO <- select(org.Hs.eg.db, 
                columns = c("GO", "SYMBOL"),
                keys = rownames(thca.filtered),
                keytype = "ENTREZID")
```

The different objects contain different information: `annot` contains the Symbol and Ensembl identifier from the remaining genes, `all.Human.GO` is the object with Gene Ontologies and Symbols for all human genes, while `anot.Entrez.GO` has the GO and the symbols for the genes.

## Differential expression of paired data

Using the expression values with batch effect removed by qr decomposition and without the surrogate variables, we proceed with the voom normalization

```{r DE, autodep=TRUE, fig.cap="\\label{voom} Figure 21: The effect of the normalization by voom."}
# Transform the log2 normalized by cqn to counts without the surrogate variables
n_counts <- apply(e.no_surrogates, 1:2, function(x){2^(x)})

# Normalize by mean variance with voom
vo <- voom(n_counts, design, plot = TRUE)
fit <- lmFit(vo, design)

# Make the comparisons we are interested in based on our design
cm <- makeContrasts(
    FemaleTumorVsFemaleNormal = tumor.FEMALE - normal.FEMALE,
    MaleTumorVsMaleNormal = tumor.MALE - normal.MALE,
    FemaleTumorVsMaleTumor = tumor.FEMALE - tumor.MALE,
    FemaleNormalVsMaleNormal = normal.FEMALE - normal.MALE,
    TumorVsNormal = (tumor.MALE + tumor.FEMALE) - (normal.MALE + normal.FEMALE),
    FemaleVsMale = (tumor.FEMALE + normal.FEMALE) - (tumor.MALE + normal.MALE),
    levels = design)
cm
fit2 <- contrasts.fit(fit, cm)
fit2 <- eBayes(fit2)

# Decide the threshold of differentially expressed genes
DE_threshold <- 4
lfc_threshold <- log2(DE_threshold)
res <- decideTests(fit2, lfc = lfc_threshold)
summary(res)
```

We decided a threshold of differentially expressed genes of 4 (thus log2 of it is 2), meaning that it must be 4 times more expressed in one group than in other in orther to pass the filter.
Based on these we are only interested in four contrast which will be annotated with symbol and ensembl id:

```{r tt, autodep=TRUE}
tt_tumor <- topTable(fit2, number = Inf, coef = "TumorVsNormal", p.value = 0.05)
tt_gender <- topTable(fit2, number = Inf, coef = "FemaleVsMale", p.value = 0.05)
tt_females <- topTable(fit2, number = Inf, coef = "FemaleTumorVsFemaleNormal", p.value = 0.05)
tt_males <- topTable(fit2, number = Inf, coef = "MaleTumorVsMaleNormal", p.value = 0.05)
    
# Annotate with the gene symbol
tt_tumor <- merge(tt_tumor, annot, by.x = 0, by.y = "ENTREZID", 
                  all.x = T, all.y = F)
tt_males <- merge(tt_males, annot, by.x = 0, by.y = "ENTREZID", 
                  all.x = T, all.y = F)
tt_gender <- merge(tt_gender, annot, by.x = 0, by.y = "ENTREZID", 
                   all.x = T, all.y = F)
tt_females <- merge(tt_females, annot, by.x = 0, by.y = "ENTREZID", 
                   all.x = T, all.y = F)
head(tt_tumor)
```

### Accuracy 

Diagnostics plot:

```{r accuracy,  fig.cap="\\label{accuracy}Figure 22.A: P-value distribution and comparison with the t-student distribution in the comparison Males vs Females.", autodep=TRUE}
pars <- par(mfrow = c(1, 2), mar = c(4, 5, 2, 2))
hist(tt_gender$P.Value, xlab = "Raw P-values", main = "", las = 1)
qqt(fit2$t[, 6], df = fit2$df.prior + fit2$df.residual[6], main = "", pch = ".", cex = 3)
qqline(fit2$t[, 6], lwd = 2, col = "red")
abline(0, 1, lwd = 2)
mtext("Comparing genes between females and males patients", 
      outer = TRUE, cex = 1.5, side = 3, line = -2)
```

```{r accuracy2, fig.cap="\\label{accuracy2}Figure 22.B: P-value distribution and comparison with the t-student distribution in the comparison Tumoral and Healthy samples.", autodep=TRUE}
pars <- par(mfrow = c(1, 2), mar = c(4, 5, 2, 2))
hist(tt_tumor$P.Value, xlab = "Raw P-values", main = "", las = 1)
qqt(fit2$t[, 5], df = fit2$df.prior + fit2$df.residual[5], main = "", pch = ".", cex = 3)
qqline(fit2$t[, 5], lwd = 2, col = "red")
abline(0, 1, lwd = 2)
mtext("Comparing genes between healthy and tumoral patients", 
      outer = TRUE, cex = 1.5, side = 3, line = -2)
```

```{r accuracy3, fig.cap="\\label{accuracy3}Figure 22.C: P-value distribution and comparison with the t-student distribution in the female comparison tumoral and healthy samples.", autodep=TRUE}
pars <- par(mfrow = c(1, 2), mar = c(4, 5, 2, 2))
hist(tt_females$P.Value, xlab = "Raw P-values", main = "", las = 1)
qqt(fit2$t[, 1], df = fit2$df.prior + fit2$df.residual[1], main = "", pch = ".", cex = 3)
qqline(fit2$t[, 1], lwd = 2, col = "red")
abline(0, 1, lwd = 2)
mtext("Comparing genes between females patients and healthy.", outer = TRUE, cex = 1.5, 
      side = 3, line = -2)
```

```{r accuracy4, fig.cap="\\label{accuracy4}Figure 22.D: P-value distribution and comparison with the t-student distribution in the male comparison tumoral and healthy samples.", autodep=TRUE}
pars <- par(mfrow = c(1, 2), mar = c(4, 5, 2, 2))
hist(tt_males$P.Value, xlab = "Raw P-values", main = "", las = 1)
qqt(fit2$t[, 2], df = fit2$df.prior + fit2$df.residual[2], main = "", pch = ".", cex = 3)
qqline(fit2$t[, 2], lwd = 2, col = "red")
abline(0, 1, lwd = 2)
mtext("Comparing genes between male patients and healthy.", outer = TRUE, cex = 1.5, 
      side = 3, line = -2)
```

### Volcano plots

We can see in volanco plots how look like the data

```{r volcanoplot, fig.cap="\\label{volcano1}Figure 23.A: Volcano plot of the comparison healthy vs tumoral.", autodep=TRUE}
volcano <- function(data, title){
    # Plot a volcano plot of a toptable output of limma
    ggplot(data, aes(logFC, B, colour = adj.P.Val)) + geom_point() + theme_bw()+ 
        ggtitle(title) + geom_vline(xintercept = c(2, -2), col = "red")+ 
        geom_hline(yintercept = 4.6, col = "red")
}

volcano(tt_tumor, "Comparing tumoral vs normal")
```

```{r volcano2, fig.cap="\\label{volcano2}Figure 23.B: Volcano plot of the comparison males vs females.", autodep=TRUE}
volcano(tt_gender, "Comparing females vs males")
```

```{r volcano3, fig.cap="\\label{volcano3}Figure 23.C: Volcano plot of the comparison healthy males vs tumoral males.", autodep=TRUE}
volcano(tt_males, "Comparing males: tumoral vs normal")
```

```{r volcano4, fig.cap="\\label{volcano4}Figure 23.C: Volcano plot of the comparison healthy fremales vs tumoral females.", autodep=TRUE}
volcano(tt_females, "Comparing females: tumoral vs normal")
```

### Genes differentially expressed

The number of genes which are differentialed expressed can be calculated with:

```{r de.number, autodep=TRUE}
de_tumor <- subset(tt_tumor, abs(logFC) >= lfc_threshold & B >= 4.6 )
de_gender <- subset(tt_gender, abs(logFC) >= lfc_threshold & B >= 4.6)
de_males <- subset(tt_males, abs(logFC) >= lfc_threshold & B >= 4.6)
de_females <- subset(tt_females, abs(logFC) >= lfc_threshold & B >= 4.6)

# Number of genes which pass the filter
de_genes <- c(nrow(de_tumor), nrow(de_gender), nrow(de_males), nrow(de_females))
names(de_genes) <- c("Tumor vs Normal", "Females vs Males", "Males", "Females")
de_genes
```

As we can see there aren't any gene of the  comparison male vs female that pass the filter.
We can compare if in the different comparisons such genes remain allways up-regulated or down-regulated. 

#### Exploration of differential expressed genes

But we create some function to make it easier to compare several lists:

```{r up_down, autodep=TRUE}
up_down <- function(data,  names = NULL){
    # Classify the genes in up or down, names are the names of the genes and 
    # should be on the same order
    dif_genes <- ifelse(data$logFC > 0, "up", "down")
    if (is.null(names)){
        names(dif_genes) <- rownames(data)
    } else {
        names(dif_genes) <- names
    }
    
    dif_genes
}

classify <- function(regulated1, regulated2){
    # Check how much up regulated genes in one comparison are down regulated in others

    inter <- intersect(names(regulated1), names(regulated2))
    
    genes <- c()
    genes2 <- c()
    
    # Reorder them by the name for the comparison with table
    for (name in inter){
        genes <- c(genes, regulated1[name])
        genes2 <- c(genes2, regulated2[name])
    }
    
    # Perform the comparison
    table(genes, genes2, 
          dnn = c(deparse(substitute(regulated1)), deparse(substitute(regulated2))))
}

genes_names <- function(regulated1, regulated2){
    # Extract the names of those up-regulated, down-regulated or not clear in
    # both lists
    inter <- intersect(names(regulated1), names(regulated2))
    
    up <- c()
    down <- c()
    not_clear <- c()
    for (name in inter){
        if (regulated1[name] == regulated2[name]){
            if (regulated1[name] == "up"){
                up <- c(up, name)
            } else{
                down <- c(down, name)
            }
        } else{
            not_clear <- c(not_clear, name)
        }
    }
    
    list(up = up, down = down, not_clear = not_clear)
}
```

The function `up_down` classify the genes in up-regulated and down-regulated. The function `classify` given two lists of genes up or down regulates, find the genes common in them and check if in both lists are up-regulated or down-regulated, however it doesn't display the names, just the numbers. However the function `genes_names` classify the genes on the three elements of the list: up if in both sets is up-regulated, down if in both sets the genes is down-regulated and not_clear if in a dataset the gene is up-regulated and in the other is down-regulated. We can see here the results of several comparisons:

```{r compares, autodep=TRUE}
r_tumor <- up_down(de_tumor)
r_females <- up_down(de_females)
r_males <- up_down(de_males)

# We see how do they overlap, more up-regulated, or more dow-regulated genes.
classify(r_tumor, r_males)
classify(r_tumor, r_females)
classify(r_females, r_males)

# We collect the name of genes for further studies
genes_females_males <- genes_names(r_females, r_males)
genes_tumor_males <- genes_names(r_tumor, r_males)
genes_tumor_females <- genes_names(r_tumor, r_females)
```

We can observe that there are `r length(genes_tumor_males$not_clear)` genes which depending on the comparison they change the direction from up-regulated to dow-regulated or vice versa. We have already obtained list of genes classified by the behavior in two comparisons with genes_names function. Furthermore we can extract all the names of the other comparisons to see how much they overlap:

```{r names_overlap, autodep=TRUE}
DE_tumor <- unique(rownames(de_tumor))
DE_males <- unique(rownames(de_males))
DE_females <- unique(rownames(de_females))
DE_gender <- unique(rownames(de_gender))
```

#### Venn diagrams

To visualize it we plot them with venn diagrams:

```{r venn_plots, message=FALSE, warning=FALSE, fig.cap="\\label{venn1}Figure 24.A: Representation of the number of genes differentially expressed in comparison between females, and healthy vs tumoral.", autodep=TRUE}
library("Vennerable")
plot(Venn(list(DE_tumor, DE_females),
              SetNames = c("Tumoral", "Females")))
```

```{r venn2, fig.cap="\\label{venn2}Figure 24.B: Representation of the number of genes differentially expressed in comparison between males, and healthy vs tumoral.", autodep=TRUE}
plot(Venn(list(DE_tumor, DE_males),
              SetNames = c("Tumoral", "Males")))
```

```{r venn3, fig.cap="\\label{venn3}Figure 24.C: Representation of the number of genes differentially expressed in comparison between males and females.", autodep=TRUE}
plot(Venn(list(DE_females, DE_males),
              SetNames = c("Females", "Males")))
```

As we can observe of the comparison, many genes differentially expressed on females comparing healthy samples and tumoral ones are not found on males.

## Functional enrichment

We can perform a functional enrichment on our data with the GO already stored

```{r fun.enrich.params_tumors, message=FALSE, warning=FALSE, autodep=TRUE}
library(GOstats)

go_obj <- function(genes_id, universe = unique(annot.Entrez.GO)){
    # Create an object for GOstat from ENTREZIDs
    new("GOHyperGParams", geneIds = genes_id,
                    universeGeneIds = universe,
                    annotation = "org.Hs.eg.db", ontology = "BP",
                    pvalueCutoff = 0.05, testDirection = "over", 
                    conditional = TRUE)
}

params_tumor <- go_obj(DE_tumor)
params_males <- go_obj(DE_males)
params_females <- go_obj(DE_females)
parmas_gender <- go_obj(DE_gender)

# The set of differentially expressed only in females
mask <- DE_females %in% intersect(DE_males, DE_females)
DE_females_exc <- DE_females[!mask]
DE_females_exc <- DE_females_exc[!DE_females_exc %in% intersect(DE_females_exc, DE_tumor)]
params_female_exc <- go_obj(DE_females_exc)

# The set of differentially expressed only in males
mask <- DE_males %in% intersect(DE_males, DE_females)
DE_males_exc <- DE_males[!mask]
DE_males_exc <- DE_males_exc[!DE_males_exc %in% intersect(DE_males_exc, DE_tumor)]
# params_male_exc <- go_obj(DE_males_exc)

# We explore those genes that are shared among the comparisons
# go_females_males_up <- go_obj(genes_females_males$up)
go_females_males_down <- go_obj(genes_females_males$down)
go_females_males_not_clear <- go_obj(genes_females_males$not_clear)

# go_tumor_females_not_clear <- go_obj(genes_tumor_females$not_clear)
# go_tumor_females_up <- go_obj(genes_tumor_females$up) # Genes not in the universe
go_tumor_females_down <- go_obj(genes_tumor_females$down)

go_tumor_males_not_clear <- go_obj(genes_tumor_males$not_clear)
# go_tumor_males_up <- go_obj(genes_tumor_males$up)
go_tumor_males_down <- go_obj(genes_tumor_males$down)
```

Once prepared with the object we can run the analysis:

```{r fun.enrich.tumor_tumors, message=FALSE, warning=FALSE}
go_analisys <- function(go_obj, annots.GO, size = 5, count = 5, pvalue = 0.05){
    hgOver <- hyperGTest(go_obj)
    
    GO <- GOstats::summary(hgOver)
    
    GO <- subset(GO, Size >= size & Count >= count & Pvalue <= pvalue)
    if (nrow(GO) == 0){
        warning("Not significant GO on this group with these settings.")
        return(NULL)
    }
    
    # Visualize the data
    plotgo <- ggplot(GO, aes(Size, Count, size = OddsRatio, colour = Pvalue)) + 
        geom_point() + b + ggtitle(deparse(substitute(go_obj)))
    
    GO <- GO[order(GO$OddsRatio, decreasing = TRUE), ]
    resume <- function(id, annot){
        #
        a <- unique(annot[annot$ENTREZID %in% id, "SYMBOL"])
        if (length(a) >= 1){
            return(a)
        } else{
            return("")
        }
    }
    geneIDs <- geneIdsByCategory(hgOver)[GO$GOBPID]
    geneSYMs <- sapply(geneIDs, resume, annot = annots.GO)
    if (is.matrix(geneSYMs)){
        geneSYMs <- apply(geneSYMs, 2, paste, collapse = ", ")
    } else {
        geneSYMs <- sapply(geneSYMs, paste, collapse = ", ")
    }
    GO <- merge(GO, geneSYMs, by.x = "GOBPID", by.y = 0)
    rownames(GO) <- 1:nrow(GO)
    colnames(GO) <- c(colnames(GO)[-ncol(GO)], "Genes")
    list(GO = GO, plot = plotgo)
}
```

```{r fun.enrich.t, fig.cap="\\label{tumor1}Figure 25.A: Representation of the P-value and effect of the GO of the comparison of healty and tumoral samples.", autodep=TRUE}

GO_tumor <- go_analisys(params_tumor, annot.Entrez.GO)
if (!is.null(GO_tumor)){
    print(head(GO_tumor$GO))
    plot(GO_tumor$plot)
}
```

We should do the same if we had some genes differentially expressed on the comparison between gender.

```{r fun.enrich.f, echo=TRUE, message=FALSE, fig.cap="\\label{tumor2}Figure 25.B: Representation of the P-value and effect of the GO of the genes with the differentially expressed genes in females comparison", autodep=TRUE}
GO_females <- go_analisys(params_females, annot.Entrez.GO)
if (!is.null(GO_females)){
    print(head(GO_females$GO))
    plot(GO_females$plot)
}
```

```{r fun.enrich.m, echo=TRUE, message=FALSE, fig.cap="\\label{tumor2}Figure 25.C: Representation of the P-value and effect of the GO of the genes with the differentially expressed genes in males comparison", autodep=TRUE}
GO_males <- go_analisys(params_males, annot.Entrez.GO)
if (!is.null(GO_males)){
    print(head(GO_males$GO))
    plot(GO_males$plot)
}
```


```{r fun.enrich.g, echo=TRUE, message=FALSE, fig.cap="\\label{tumor3}Figure 25.D: Representation of the P-value and effect of the GO of the genes with the differentially expressed genes in females againts males", autodep=TRUE}
GO_gender <- go_analisys(parmas_gender, annot.Entrez.GO)
if (!is.null(GO_gender)){
    print(head(GO_gender$GO))
    plot(GO_gender$plot)
}
```

```{r fun.enrich.gender_tumors, echo=TRUE, message=FALSE, fig.cap="\\label{tumor4}Figure 25.E: Representation of the P-value and effect of the GO of the comparison of differentially expressed genes exclusivly from females", autodep=TRUE}
GO_female_exc <- go_analisys(params_female_exc, annot.Entrez.GO)
if (!is.null(GO_female_exc)){
    print(head(GO_female_exc$GO))
    plot(GO_female_exc$plot)
}
```


```{r fun.enrich., eval=FALSE, message=FALSE, fig.cap="\\label{tumor5}Figure 25.F: Representation of the P-value and effect of the GO of the comparison of differentially expressed genes exclusivly from males", autodep=TRUE}
GO_male_exc <- go_analisys(params_male_exc, annot.Entrez.GO)
if (!is.null(GO_male_exc)){
    print(head(GO_male_exc$GO))
    plot(GO_male_exc$plot)
}
```


```{r fun.enrich.fmu, eval=FALSE, message=FALSE, warning=FALSE, fig.cap="\\label{tumor6}Figure 25.G: Representation of the P-value and effect of the GO of the up-regulated genes in both  male and female comparisons", autodep=TRUE}
GO_females_males_up <- go_analisys(go_females_males_up, annot.Entrez.GO)
if (!is.null(GO_females_males_up)){
    print(head(GO_females_males_up$GO))
    plot(GO_females_males_up$plot)
}
```

```{r fun.enrich.fmd, echo=TRUE, message=FALSE, warning=FALSE, fig.cap="\\label{tumor7}Figure 25.H: Representation of the P-value and effect of the GO of the down-regulated genes in both  male and female comparisons", autodep=TRUE}
GO_females_males_down <- go_analisys(go_females_males_down, annot.Entrez.GO)
if (!is.null(GO_females_males_down)){
    print(head(GO_females_males_down$GO))
    plot(GO_females_males_down$plot)
}
```


```{r fun.enrich.tfn, eval=FALSE, message=FALSE, fig.cap="\\label{tumor8}Figure 25.I: Representation of the P-value and effect of the GO of the genes with different regulation in the comparison between normal and tumors and females", autodep=TRUE}
GO_tumor_females_not_clear <- go_analisys(go_tumor_females_not_clear, annot.Entrez.GO)
if (!is.null(GO_tumor_females_not_clear)){
    print(head(GO_tumor_females_not_clear$GO))
    plot(GO_tumor_females_not_clear$plot)
}
```

```{r fun.enrich.tfu, eval=FALSE, message=FALSE, fig.cap="\\label{tumor9}Figure 25.J: Representation of the P-value and effect of the GO of the genes up-regulated in both comparisons between normal and tumors and tumor vs normal in females.", autodep=TRUE}
GO_tumor_females_up <- go_analisys(go_tumor_females_up, annot.Entrez.GO)
if (!is.null(GO_tumor_females_up)){
    print(head(GO_tumor_females_up$GO))
    plot(GO_tumor_females_up$plot)
}
```

```{r fun.enrich.tfd, echo=TRUE, message=FALSE, fig.cap="\\label{tumor10}Figure 25.K: Representation of the P-value and effect of the GO of the genes down-regulated in both comparisons between normal and tumors and tumor vs normal in females.", autodep=TRUE}
GO_tumor_females_down <- go_analisys(go_tumor_females_down, annot.Entrez.GO)
if (!is.null(GO_tumor_females_down)){
    print(head(GO_tumor_females_down$GO))
    plot(GO_tumor_females_down$plot)
}
```


```{r fun.enrich.tmn, echo=TRUE, message=FALSE, fig.cap="\\label{tumor11}Figure 25.L: Representation of the P-value and effect of the GO of the genes with the differentially expressed genes in males againts tumor comparison", autodep=TRUE}
GO_tumor_males_not_clear <- go_analisys(go_tumor_males_not_clear, annot.Entrez.GO)
if (!is.null(GO_tumor_males_not_clear)){
    print(head(GO_tumor_males_not_clear$GO))
    plot(GO_tumor_males_not_clear$plot)
}
```

```{r fun.enrich.tmu, eval=FALSE, message=FALSE, fig.cap="\\label{tumor12}Figure 25.M: Representation of the P-value and effect of the GO of the genes with the differentially expressed genes up-regulated in both tumors vs normal in males and tumor vs normals.", autodep=TRUE}
GO_tumor_males_up <- go_analisys(go_tumor_males_up, annot.Entrez.GO)
if (!is.null(GO_tumor_males_up)){
    print(head(GO_tumor_males_up$GO))
    plot(GO_tumor_males_up$plot)
}
```

```{r fun.enrich.tmd, echo=TRUE, message=FALSE, fig.cap="\\label{tumor13}Figure 25.N: Representation of the P-value and effect of the GO of the genes down-regulated in both comparisons between normal and tumors and tumor vs normal in males.", autodep=TRUE}
GO_tumor_males_down <- go_analisys(go_tumor_males_down, annot.Entrez.GO)
if (!is.null(GO_tumor_males_down)){
    print(head(GO_tumor_males_down$GO))
    plot(GO_tumor_males_down$plot)
}
```

### Result tables

We should look in details to those values and associations in the tables.

```{r tables_GO, message=FALSE, autodep=TRUE}
library(xtable)
table_h <- function(go_obj, ftitle){
    # Function to output the table in html format
    if (is.null(go_obj)){
        warning("Null Object: unable to creat the table")
    } else {
        xtab_tumor <- xtable(go_obj, align="l|c|r|r|r|r|r|p{3cm}|p{3cm}")
        print(x=xtab_tumor, file = ftitle, type = "html")
    }
}

table_g <- function(go_obj, ftitle){
    # Function to output the table in html format
    if (is.null(go_obj)){
        warning("Null Object: unable to creat the table")
    } else {
        xtab_tumor <- xtable(go_obj, align="l|c|r|r|r|r|r|p{3cm}|p{3cm}|r|")
        print(x=xtab_tumor, file = ftitle, type = "html")
    }
}
table_h(GO_tumor$GO, "GO_tumor_table.html")
table_h(GO_females$GO, "GO_females_table.html")
# table_h(GO_males$GO, "GO_males_table.html")
table_h(GO_gender$GO, "GO_gender_table.html")

table_h(GO_female_exc$GO, "GO_female_exc_table.html")
# table_h(GO_male_exc$GO, "GO_male_exc_table.html")

# table_h(GO_females_males_up$GO, "GO_females_males_up_table.html")
table_h(GO_females_males_down$GO, "GO_females_males_down_table.html")
# table_h(GO_females_males_not_clear$GO, "GO_females_males_not_clear_table.html")

# table_h(GO_tumor_females_not_clear$GO, "GO_tumor_females_not_clear_table.html")
table_h(GO_tumor_males_not_clear$GO, "GO_tumor_males_not_clear_table.html")
# table_h(GO_tumor_females_up$GO, "GO_tumor_females_up_table.html")
# table_h(GO_tumor_males_up$GO, "GO_tumor_males_up_table.html")
table_h(GO_tumor_females_down$GO, "GO_tumor_females_down_table.html")
table_h(GO_tumor_males_down$GO, "GO_tumor_males_down_table.html")

table_g(de_tumor, "DE_tumor.html")
table_g(de_gender, "DE_gender.html")
table_g(de_females, "DE_females.html")
table_g(de_males, "DE_males.html")
```

The table of genes which are differentially expressed genes are :
The comparison of [tumoral vs normal](DE_tumor.html) samples.  
The comparison of [females vs males](DE_gender.html) samples.  
The comparison of [tumoral vs normal](DE_females.html) female samples.  
The comparison of [tumoral vs normal](DE_males.html) male samples.  

To explore more than the top 6 GO we should look into the tables generated with any browser.
In [GO_tumor_table](GO_tumor_table.html) we have the list of GO enrich on the differentially expressed genes in the comparison tumor vs healthy samples.  
In [GO_females_table](GO_females_table.html) we have the list of GO enrich on the differentially expressed genes in the comparison tumoral females vs healthy females.  
In [GO_males_table](GO_males_table.html) we have the list of GO enrich on the differentially expressed genes in the comparison between tumoral males samples and healthy males samples.  
In [GO_female_exc](GO_female_exc_table.html) we have the list of GO enrich on the differentially expressed genes in the comparison tumor vs normals of females, exclusively from females.  
In [GO_male_exc](GO_male_exc_table.html) we have the list of GO enrich on the differentially expressed genes in the comparison male tumor vs normal samples, exclusively from males.  
In [GO_females_males_up](GO_females_males_up_table.html) we have the list of GO enrich on the differentially expressed genes which are shared between the males and the females and in both cases are up-regulated.  
In [GO_females_males_down](GO_females_males_down_table.html) we have the list of GO enrich on the differentially expressed genes which are shared between the males and the females and in both cases are down-regulated.  
In [GO_tumor_females_not_clear](GO_tumor_females_not_clear_table.html) we have the list of GO enrich on the differentially expressed genes shared between the comparison of tumoral vs healthy samples and tumoral samples and healthy females.  
In [GO_tumor_males_not_clear](GO_tumor_males_not_clear_table.html) we have the list of GO enrich on the differentially expressed genes shared between the comparison of tumoral vs healthy samples and tumoral samples and healthy males.   
In [GO_tumor_females_up](GO_tumor_females_up_table.html) we have the list of GO enrich on the differentially expressed genes up-regulated in the comparison of tumoral vs healthy samples and tumoral samples and healthy females.  
In [GO_tumor_males_down](GO_tumor_males_down_table.html) we have the list of GO enrich on the differentially expressed genes down-regulated in the comparison of tumoral vs healthy samples and tumoral samples and healthy males. 
In [GO_tumor_females_down](GO_tumor_females_down_table.html) we have the list of GO enrich on the differentially expressed genes down-regulated in the comparison of tumoral vs healthy samples and tumoral samples and healthy females.  
In [GO_tumor_males_up](GO_tumor_males_up_table.html) we have the list of GO enrich on the differentially expressed genes up-regulated in the comparison of tumoral vs healthy samples and tumoral samples and healthy males.  

## SessionInfo

```{r sessionInfo}
sessionInfo()
```


##Bibliography

* Agrawal N, Akbani R, Aksoy BA, Ally A, Arachchi H, Asa SL, Zou L. (2014). Integrated Genomic Characterization of Papillary Thyroid Carcinoma. Cell, 159(3), 676-690. 
* The Canger Genome Atlas. Papillary Thyroid Carcinoma. Retrieved May 6, 2016. Available from: http://cancergenome.nih.gov/cancersselected/thyroid
* Xing M. (2013). Molecular pathogenesis and mechanisms of thyroid cancer. Nature Reviews. Cancer, 13(3), 184-99. 
