\documentclass[9pt,twocolumn,twoside]{gsajnl}
% Use the documentclass option 'lineno' to view line numbers

\articletype{inv} % article type
% {inv} Investigation 
% {gs} Genomic Selection
% {goi} Genetics of Immunity 
% {gos} Genetics of Sex 
% {mp} Multiparental Populations
\usepackage{cite}
\usepackage{url}


\title{New insights of the RNA-seq data of \textit{The Cancer Genome Atlas} for Papillary Thyroid Carcinoma}

\author[$\ast$,1]{Luís Revilla}
\author[$\ast$]{Marina Reixachs}
\author[$\ast$]{Mauro Álvarez}
\author[$\ast$]{Inés Sentís}


\affil[$\ast$]{Universitat Pompeu Fabra}


\keywords{Cancer; Genomics; Thyroid; ...}

\runningtitle{GENETICS Journal Template on Overleaf} % For use in the footer 

\correspondingauthor{Corresponding Author}

\begin{abstract}
The abstract should be written for people who may not read the entire paper, so it must stand on its own. The impression it makes usually determines whether the reader will go on to read the article, so the abstract must be engaging, clear, and concise. In addition, the abstract may be the only part of the article that is indexed in databases, so it must accurately reflect the content of the article. A well-written abstract is the  most effective way to reach intended readers, leading to more robust search, retrieval, and usage of the article. 

Please see additional guidelines notes on preparing your abstract below.
\end{abstract}

\setboolean{displaycopyright}{true}

\begin{document}

\maketitle
\thispagestyle{firststyle}
\marginmark
\firstpagefootnote
\correspondingauthoraffiliation{Universitat Pompeu Fabra. Doctor Aiguader, 80. 08003 Barcelona (Spain). 
Email: lluis.revilla01@estudiant.upf.edu}
\vspace{-11pt}%


\section*{Introduction}

Thyroid cancer can develop from the different cells that form the follicles of the thyroid. This gland located at the base of the throat secretes hormones such as T3 and T4 that have their metabolic functionalities such as control of heart rate, blood pressure, body temperature and weight (see it on \href{http://cancergenome.nih.gov/cancersselected/thyroid}{ TCGA}).

From the four types of thyroid cancer (papillary, follicular, medullary, and anaplastic thyroid cancer) Papillary Thyroid Carcinoma (PTC) is the most common type of thyroid cancer \citep{Agrawal2014a}. It is more prevalent in women than men and its common diagnosis occurs between 25 and 65 years old (see it \href{http://www.cancer.org/cancer/thyroidcancer/detailedguide/thyroid-cancer-risk-factors}{here}).

Driver onco-mutations for this type of cancer appear as alterations of the MAPK signaling pathway and the PI3K-AKT pathway \citep{Kimura2003}. Both are implied in cell proliferation and survival and in human tumorigenesis. The overactivation of the MAPK pathway because of mutations such as the BRAFV600E mutation, leads to the development of papillary thyroid cancer (PTC) from follicular thyroid cells \citep{Xing2013a}. On the other hand, mutations that activate the PI3K-AKT pathway, such as mutations in RAS, PTEN and PIK3CA, mostly leads to development of follicular thyroid adenoma (FTA) and follicular thyroid cancer (FTC) also in follicular thyroid cells \citep{Xing2013a}.

Research from \href{http://cancergenome.nih.gov/cancersselected/thyroid}{ The Cancer Genome Atlas} focused on PTC and they found that a part from alterations in BRAF (specifically BRAFV600E) and RAS, there are other driver mutations such the ones in EIF1AX PPM1D, and CHEK2 genes that are main alterations for that type of thyroid cancer . They proposed a new classification of PTC subtypes based on their molecular findings. 

The aim of our project deviates from being a replicate of the TCGA project. Our academic purpose has been to study differentially expressed genes from tumor and normal samples but also taking into account whether there could be a gender effect on the tumorgenesis of PTC. To our knowledge, there is no clear reason why thyroid cancer is more prevalent in women than men. Therefore, it would be interesting to analyze whether there are differentially expressed genes within genders and respect to thyroid cancer. In addition, in case there are significant differences in gene expression which are the main pathways affected. 


\section*{Materials and Methods}

\subsection*{Data Availability}
In order to shed light on the previous hypothesis we have centered our study in Papillary Thyroid Carcinoma with RNA-seq data of The Cancer Genome Atlas \citep{tcga}. Concretely, the datasets used have been the ones preprocessed by \citet{Rahman2015} using the \citep{subread} pipeline. However, although the data has been preprocessed we have had to apply different filters to speed up the analysis. While exploring the data we realized that all 'normal' type samples came from  individuals who had 'tumor' type samples too so we only kept values coming from paired data (see Supplementary Materials). In addition, from the subset we discarted those samples whose depth is substantially lower than the rest specifically below 40 million reads (see Supplementary Materials). Moreover, we have considered a cutoff of 1 log CPM unit as minimum value of expression to select genes being expressed across samples. Using this cutoff we have filtered out lowly-expressed genes (see Supplementary Materials). Finally,the most abundant population in the samples is the "white" and "not hispanic or latino", so we decide to perform the analysis on this group of individuals whose type of cancer is the "classical" Papillary Thyroid Carcinoma (see Supplementary Materials). The final subset for the analysis end up with 11801 genes, 54 samples (27 normal and 27 tumor) from 27 patients. To reduce the size of the subset we also discarded clinical variables without available information or under 60 \% of the information available (see Supplementary Materials). 

\subsection*{Normalization and Quality Assessment}
All the analysis presented in this study has been done using the R statistical programming language \citep{R}. Data has been managed within \textit{SummarizedExperiment} class object container \citep{SummarizedExperiment} of the Bioconductor open-source and open-development software for R \citep{Gentleman2004}. Before any analysis, we have normalized the data following the edgeR pipeline for normalizing RNA-seq data \citep{Robinson2010}. Robinson's guide consist on first of all scaling the data into \textit{counts per million reads} (CPM). Then we have checked for biases in the data with MA plots (see Supplementary Materials). From the MA plots we decided to discard tumor and normal samples having  TCGA-BJ-A3PR as barcode because we detected a relevant expression-level dependent bias so the subset was reduced to 52 samples (26 tumor and 26 normal) from 26 patients. After that, we also have perfomed a quantile normalization on the RNA-seq count data using the \textit{cqn} R/Bioconductor package \citep{Hansen2012}. We decided not to use the function \textit{calcNormFactors} which has implemented the \textit{Trimmed Mean of M-values} (TMM) method of the \textit{edgeR} R/Bioconductor package \citep{Robinson2010a} because according to the \textit{cqn} vignette, there is no need to normalize using this type of tools as later we used the offsets computed by \textit{cqn} \citep{Hansen2012}. 

\subsection*{Batch effects assessment}
A part from normalization, we also have considered possible batch effects. We have used the information of the TCGA barcode included in the sample identification as surrogates for batch effect indicators following the MD Anderson Cancer Center strategy \citep{batch}. We looked at the cross-classification of our outcome of interest (normal vs. tumor) and the candidate indicators of batch effect: samplevial, tss, 
portion and plate (see Supplementary Materials). A part from that, we also have used hierarchical clustering of the samples calculating the distances between each pair of samples with the Spearman correlation and then using the \textit{hclust} function of R. To further investigate whether batch might be a source of variation we perfomed a mutidimensional scaling using the function \textit{plotsMDS} of the Bioconductor/R package \textit{edgeR} \citep{Robinson2010}. After identification of the batch effect (see Supplementary Materials), we have used different strategies to remove it. First, we tried with the function \textit{ComBat} of the Bioconductor/R \textit{sva} package \citep{sva}. We also have used QR descomposition technique implemented in the \textit{limma} Bioconductor/R package \citep{limma} through the function \textit{removeBatchEffect}. Third strategy used has been the SVD method \citep{Alter2000} which was implemented in the Bioconductor/R package \textit{corpcor} \citep{corpcor} with the function \textit{fast.svd}.

\subsection*{Differential expression analysis}
In order to perform the differential expression analysis we used \textit{limma} R/Bioconductor package with a workflow based on what has been previously published \citep{limma}. We build the design matrix and calculated the observational-level weights. As we were working with paired data we estimated the correlation in repeated measurements before fitting the linear model. Finally we build and fitted contrast matrices related to gender and disease status (normal or tumor) and calculated moderated t-statistics. Moreover we used the \textit{sva} R/Bioconductor package to remove the effect of the surrogate variables \citep{sva}. Representation of differentially expressed gene sets was performed with the R package \textit{Vennerable} \citep{Venn}.

\subsection*{Functional enrichment}
The resulting sets of differentially expressed genes between different conditions according to gender or disease status (normal or tumor) was used to perform Gene Ontology (GO) analyses. For this purpose we used the R/Bioconductor package \textit{GOstats} with the Biological Process (BP) ontology and applied a conditional hypergeometric test in order to account for the hierarchical structure of GO terms \citep{Gostats}.


\section*{Results and Discussion}

After the filtering and normalization of the data, we searched for batch effects. Apparently, after applying hierarchical clustering in the filtered data the samples cluster into tumor and normal separately (see Supplementary Materials). However, from the dendrogram we can see that there is a group of tumor samples closer to the normal ones as if there were two subgroups of tumor samples. Cross-tables between the elements encoded in the TCGA barcode and type of samples (tumor/normal) showed unbalanced data in all of them so it could be that one of them is the source of variability within tumor samples. From the dendrogram plots after clustering for each of the elements (samplevial, tss, portion and plate) and also after considering a multidimensional scaling plot it did not seem clear to us which was the reason for such inaccurate clustering of the tumor samples (see Supplementary Materials). We decided to apply the three strategies mentioned before (see Materials and Methods) to see whether we could improve the clustering. Although, \textit{ComBat} function and the SVD method are supposed to improved the clustering, the one that gave us a better dendrogram has been QR descomposition method (see Supplementary Materials). For this removal strategy, we included two batch indicator variables, tss and samplevial, as the possible sources of variability and we also built the full model of our outcome of interest which included gender and type as fixed effects. As it has yield the best batch removal result, we decided to take it into account in the follow-up analysis. 

A part from removing possible batch effect, we also have adjusted for the mean-variance relationship estimating weights at individual observation-by-gene level to fit the model. Furthermore, we have also adjusted for possible intra group correlation at individual level (blocking factor: \textit{bcr\_patient\_barcode}) which we included as a factor into the model. 

From the model, we were trying to figure out whether there was a gender effect in the expression level of certain genes in the context of thyroid tumor. So we were interested to see whether DE genes observed between tumor and normal were the same for within female samples (tumor  and normal) and within male samples (tumor and normal). For this reason, our contrasts of interest were made as 4 comparisons: Male vs. Female, Disease(tumor) vs. Normal, Male-Disease vs. Male-Normal,  Female-Disease vs. Female-Normal. Here we show the diagnostic plots for those comparisons (see Figure)...

We can see from the QQplots that DE genes from the comparison Disease vs. Normal differ from adjusting to a t-Student distribution (null hypothesis/no-DE) more than the gender comparison (Males vs Females). We can see it is also reflected within genders in the Disease-Female vs. Normal-Female and Disease-Male vs. Normal-Male comparisons. The volcano plots (Figure...) reveal that DE genes within genders are more due to the disease itself rather than a gender effect. In other words, disease condition (tumor or normal) is a much better predictor of DE genes than gender. Concretely, there are 714 DE genes between normal and tumor samples, 2 DE genes between female and male samples, 72 DE genes between Male-Normal vs. Male-Disease and 116 DE genes between Female-Normal vs. Female-Disease.

Venn diagrams in Figure ... help to explore the consistency of the obtained DE genes among the different comparisons. From the set of DE genes between disease and normal condition only 3 are found to be also differentially expressed in females (Female-Normal vs. Female-Disease) and 2 in males (Male-Normal vs. Male-Disease). So, even though the pattern from the volcano plots and the QQ plots of the 3 comparisons were similar actually a small amount of DE genes are shared to be differentially expressed among them. In contrast, taking both within gender comparisons (Female-Normal vs. Female-Disease and Male-Normal vs. Male-Disease) a larger proportion of genes are shared (54). Furhtermore, the direction of the fold change is also consistent in both comparisons, giving a set of 26 genes that are down regulated and 28 up regulated.





  




\section*{Additional guidelines}

\subsection*{Numbers} In the text, write out numbers nine or less except as part of a date, a fraction or decimal, a percentage, or a unit of measurement. Use Arabic numbers for those larger than nine, except as the first word of a sentence; however, try to avoid starting a sentence with such a number.

\subsection*{Units} Use abbreviations of the customary units of measurement only when they are preceded by a number: "3 min" but "several minutes". Write "percent" as one word, except when used with a number: "several percent" but "75\%." To indicate temperature in centigrade, use ° (for example, 37°); include a letter after the degree symbol only when some other scale is intended (for example, 45°K).

\subsection*{Nomenclature and Italicization} Italicize names of organisms even when  when the species is not indicated.  Italicize the first three letters of the names of restriction enzyme cleavage sites, as in HindIII. Write the names of strains in roman except when incorporating specific genotypic designations. Italicize genotype names and symbols, including all components of alleles, but not when the name of a gene is the same as the name of an enzyme. Do not use "+" to indicate wild type. Carefully distinguish between genotype (italicized) and phenotype (not italicized) in both the writing and the symbolism.

\section*{In-text Citations}

Add citations using the \verb|\citep{}| command, for example \citep{neher2013genealogies} or for multiple citations, \citep{neher2013genealogies, rodelsperger2014characterization}

\section*{Examples of Article Components}
\label{sec:examples}

The sections below show examples of different header levels, which you can use in the primary sections of the manuscript (Results, Discussion, etc.) to organize your content.

\section*{First level section header}

Use this level to group two or more closely related headings in a long article.

\subsection*{Second level section header}

Second level section text.

\subsubsection*{Third level section header:}

Third level section text. These headings may be numbered, but only when the numbers must be cited in the text. 

\section*{Figures and Tables}

Figures and Tables should be labelled and referenced in the standard way using the \verb|\label{}| and \verb|\ref{}| commands.

\subsection*{Sample Figure}

Figure \ref{fig:spectrum} shows an example figure.

\begin{figure}[htbp]
\centering
\includegraphics[width=\linewidth]{example-figure}
\caption{Example figure from \url{10.1534/genetics.114.173807}. Please include your figures in the manuscript for the review process. You can upload figures to Overleaf via the Project menu. Upon acceptance, we'll ask for your figure files to be uploaded in any of the following formats: TIFF (.tiff), JPEG (.jpg), Microsoft PowerPoint (.ppt), EPS (.eps), or Adobe Illustrator (.ai).  Images should be a minimum of 300 dpi in resolution and 500 dpi minimum if line art images.  RGB, CMYK, and Grayscale are all acceptable. Halftones should be high contrast with sharp detail, because some loss of detail and contrast is inevitable in the production process. Figures should be 10-20 cm in width and 1-25 cm in height. Graph axes must be exactly perpendicular and all lines of equal density.
Label multiple figure parts with A, B, etc. in bolded type, and use Arrows and numbers to draw attention to areas you want to highlight. Legends should start with a brief title and should be a self-contained description of the content of the figure that provides enough detail to fully understand the data presented. All conventional symbols used to indicate figure data points are available for typesetting; unconventional symbols should not be used. Italicize all mathematical variables (both in the figure legend and figure) , genotypes, and additional symbols that are normally italicized.  
}%
\label{fig:spectrum}
\end{figure}

\subsection*{Sample Video}

Figure \ref{video:spectrum} shows how to include a video in your manuscript.

\begin{figure}[htbp]
\centering
\includegraphics[width=\linewidth]{example-figure}
\caption{Example movie (the figure file above is used as a placeholder for this example). \textit{GENETICS} supports video and movie files that can be linked from any portion of the article - including the abstract. Acceptable formats include .asf, avi, .wav, and all types of Windows Media files.   
}%
\label{video:spectrum}
\end{figure}


\subsection*{Sample Table}

Table \ref{tab:shape-functions} shows an example table. Avoid shading, color type, line drawings, graphics, or other illustrations within tables. Use tables for data only; present drawings, graphics, and illustrations as separate figures. Histograms should not be used to present data that can be captured easily in text or small tables, as they take up much more space.  

Tables numbers are given in Arabic numerals. Tables should not be numbered 1A, 1B, etc., but if necessary, interior parts of the table can be labeled A, B, etc. for easy reference in the text.  


\begin{table*}[htbp]
\centering
\caption{\bf Students and their grades}
\begin{tableminipage}{\textwidth}
\begin{tabularx}{\textwidth}{XXXX}
\hline
Student & Grade\footnote{This is an example of a footnote in a table. Lowercase, superscript italic letters (a, b, c, etc.) are used by default. You can also use *, **, and *** to indicate conventional levels of statistical significance, explained below the table.} & Rank & Notes \\
\hline
Alice & 82\% & 1 & Performed very well.\\
Bob & 65\% & 3 & Not up to his usual standard.\\
Charlie & 73\% & 2 & A good attempt.\\
\hline
\end{tabularx}
  \label{tab:shape-functions}
\end{tableminipage}
\end{table*}

\section*{Sample Equation}

Let $X_1, X_2, \ldots, X_n$ be a sequence of independent and identically distributed random variables with $\text{E}[X_i] = \mu$ and $\text{Var}[X_i] = \sigma^2 < \infty$, and let
\begin{equation}
S_n = \frac{X_1 + X_2 + \cdots + X_n}{n}
      = \frac{1}{n}\sum_{i}^{n} X_i
\label{eq:refname1}
\end{equation}
denote their mean. Then as $n$ approaches infinity, the random variables $\sqrt{n}(S_n - \mu)$ converge in distribution to a normal $\mathcal{N}(0, \sigma^2)$.

\bibliography{IEO}

\end{document}
